phoenix-core/src/main/java/org/apache/phoenix/schema/task/Task.java

```diff
@@ -230,7 +232,20 @@ public static void deleteTask(PhoenixConnection conn, PTable.TaskType taskType,
                 taskQuery += " AND " + PhoenixDatabaseMetaData.TASK_DATA + " LIKE '%" + indexName + "%'";
             }
 
-        return populateTasks(connection, taskQuery);
+            List<TaskRecord> taskRecords = populateTasks(connection, taskQuery);
+            List<TaskRecord> result = new ArrayList<TaskRecord>();
+            if (ts != null) {
+                // Adding TASK_TS to the where clause did not work. It returns empty when directly querying with the timestamp.
+                for (TaskRecord tr : taskRecords) {
```

Comment:
@gokceni Why do we need these changes for fixing the test?

Reply:
ping @gokceni please address these review comments from earlier. Also, would this change affect any other tasks wherein timestamp is specified?

Reply:
@ChinmaySKulkarni Your previous comments about uniquely getting a task made me question about task query. I realized that there might be a case where we have tasks for the same index and table but with a different timestamp. I change the code to handle that.